pkg/cloudprovider/provider/openstack/provider.go

```diff
@@ -282,6 +283,15 @@ func getClient(c *Config) (*gophercloud.ProviderClient, error) {
 		pc.HTTPClient = cloudproviderutil.HTTPClientConfig{LogPrefix: "[OpenStack API]"}.New()
 	}
 
+	// use the util's HTTP client to benefit from its CA bundle
+	httpConfig := util.HTTPClientConfig{
+		// before machine-controller v1.25 this used the default client with no timeout,
+		// so this is deliberately set to a high value to avoid too many disruptions
+		Timeout: 60 * time.Second,
+	}
+
+	pc.HTTPClient = httpConfig.New()
```

Comment:
Here `pc.HTTPClient` is unconditionally overwritten from what might been set just few lines above. Looks wrong.

Reply:
Oh wow, yeah, that was a total brainfart.